src/tests/ftest/pool/dynamic_server_pool.py

```diff
@@ -0,0 +1,151 @@
+        new_pool.create()
+        self.expected_uuids.append(new_pool.uuid.lower())
+        self.pool.append(new_pool)
+
+    def verify_uuids(self):
+        """Call dmg pool list to get the list of UUIDs and verify against the
+        expected UUIDs passed in.
+        """
+        actual_uuids = []
+        output_list = self.get_dmg_command().get_output("pool_list")
+        for uuid_sr in output_list:
+            actual_uuids.append(uuid_sr[0])
+        actual_uuids.sort()
+        self.expected_uuids.sort()
+        self.assertEqual(self.expected_uuids, actual_uuids)
+
+    def test_dynamic_server_pool(self):
+        """
+        JIRA ID: DAOS-3598
+
+        Test Description: See class description.
+
+        :avocado: tags=all,medium,control,full_regression,dynamic_server_pool
+        """
+        self.pool = []
+
+        # Create a pool on rank0.
+        self.create_new_pool("0")
+        # Create a pool across the 2 servers.
+        self.create_new_pool()
+        # Verify UUIDs by calling dmg pool list.
+        self.verify_uuids()
+
+        # Verify that the UUID-named directory is created, or not created, at
+        # each host for the two pools.
+        # Note that there's no mapping between rank and hostname, but it appears
+        # that self.hostlist_servers[0] is always rank0, 1 is rank1, and the
+        # extra server we'll be adding will be rank2.
+        RC_SUCCESS = 0
+        RC_FAIL = 1
+        rc_p1h1 = check_for_pool(
+            self.hostlist_servers[0], self.pool[0].uuid.lower())
+        self.assertEqual(rc_p1h1, RC_SUCCESS)
+        rc_p1h2 = check_for_pool(
+            self.hostlist_servers[1], self.pool[0].uuid.lower())
+        self.assertEqual(rc_p1h2, RC_FAIL)
+        rc_p2h1 = check_for_pool(
+            self.hostlist_servers[0], self.pool[1].uuid.lower())
+        self.assertEqual(rc_p2h1, RC_SUCCESS)
+        rc_p2h2 = check_for_pool(
+            self.hostlist_servers[1], self.pool[1].uuid.lower())
+        self.assertEqual(rc_p2h2, RC_SUCCESS)
+
+        # Start an additional server.
+        extra_servers = self.params.get("test_servers", "/run/extra_servers/*")
+        self.log.info("Extra Servers = %s", extra_servers)
+        self.start_additional_servers(extra_servers)
+
+        # Create a pool on the newly added server and verify the UUIDs with dmg
+        # pool list.
+        self.create_new_pool("2")
+        self.verify_uuids()
+        # Verify that the UUID-named directory is created, or not created, at
+        # each host for the new pool.
+        rc_p3h1 = check_for_pool(
+            self.hostlist_servers[0], self.pool[2].uuid.lower())
+        self.assertEqual(rc_p3h1, RC_FAIL)
+        rc_p3h2 = check_for_pool(
+            self.hostlist_servers[1], self.pool[2].uuid.lower())
+        self.assertEqual(rc_p3h2, RC_FAIL)
+        rc_p3h3 = check_for_pool(extra_servers[0], self.pool[2].uuid.lower())
+        self.assertEqual(rc_p3h3, RC_SUCCESS)
```

Comment:
Thanks for suggestion. I noticed that just because pool.target_list.value is None, that doesn't mean the pool should exist because we add a server during the test. Instead, I created a dictionary parameter to pass in the mapping from the UUID to the expected ranks.

I agree that it's a good idea to have a function to test every UUID on every host instead of only the new pool's UUID.